tests/Avalonia.Benchmarks/Animations/TransitionBenchmark.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using System.Runtime.CompilerServices;
+using Avalonia.Animation;
+using Avalonia.Animation.Animators;
+using Avalonia.Layout;
+using BenchmarkDotNet.Attributes;
+
+namespace Avalonia.Benchmarks.Animations
+{
+    [MemoryDiagnoser]
+    public class TransitionBenchmark
+    {
+        private readonly DoubleTransition _transition;
+        private readonly DoubleTransitionOld _oldTransition;
+        private readonly Subject<double> _timeProducer;
+        private readonly List<double> _producedValues;
+        private readonly AddValueObserver _observer;
+
+        [Params(10, 100)]
+        public int FrameCount { get; set; }
+
+        public TransitionBenchmark()
+        {
+            _oldTransition = new DoubleTransitionOld
+            {
+                Duration = TimeSpan.FromMilliseconds(FrameCount), Property = Layoutable.WidthProperty
+            };
+
+            _transition = new DoubleTransition
+            {
+                Duration = TimeSpan.FromMilliseconds(FrameCount), Property = Layoutable.WidthProperty
+            };
+
+            _timeProducer = new Subject<double>();
+            _producedValues = new List<double>(FrameCount);
+
+            _observer = new AddValueObserver(_producedValues);
+        }
+
+        [Benchmark(Baseline = true)]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void OldTransition()
```

Comment:
Not sure if this old implementation benchmark makes much sense.
Most of your optimizations were applied for both of them anyway.

Reply:
Yeah, I think I will remove it now.